### ISSUE TITLE
Challenges with A/B testing CustomEditor and Notebooks

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode#100523]
We (`Python Extension`) would like to roll out the Custom Editor and Notebook to users as an A/B test.

I.e. by default our extension will not be using custom editor or notebook. But during runtime we would like to enable either one of the two for a user (based on some algorithm a user would get picked up and the feature enabled). 

To make this possible:
* We need to register the static contributions in the package.json (done with priority=option)
* Register the necessary VS Code API (providers) in code

However, we cannot have the priority of the view type in the contribution set to `default`, as we don't want everyone to get this. Hence we have two solutions:
* During runtime we modify the package.json file and prompt the user to reload VS Code (this is our current solution).
    * We're modifying the `package.json` (super hacky)
	* Not sure this would work in SSH, code spaces and similar remote scenarios.
* Or we prompt & update the users `settings.json` file to ensure all `*.ipynb` files are associated with the corresponding view.
	* We cannot make changes without the users consent.
	* Now we have yet another prompt.
	* This could defeats the purpose of the experiment, we don't want to give users a choice upfront, as users might decide not to allow changes to settings files (i wouldn't), hence skewing the A/B testing metrics.

Basically the only way we can A/B test custom editor/notebooks is to make changes to `package.json`.

Please advice if there's another solution.
Filing after discussing with @rebornix

**Comment by DonJayamanne** [comment]
Closing based on feeback from @rebornix

**Comment by rebornix** [comment]
Modifying users settings is the preferred solution here as tweaking `package.json` will require window reloading (which breaks users current workflow) and also VS Code will warn that the extension is modified on disk. Technically they are identical as none is transparent to users.
